xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -72,13 +54,28 @@ void CRendererDXVA::GetWeight(std::map<RenderMethod, int>& weights, const VideoP
                  DX::DXGIFormatToString(dxgi_format));
       return;
     }
+  }
 
-    if (enumerator.SupportedConversions(SupportedConversionsArgs{picture, systemUsesHDR}).empty())
-    {
-      CLog::LogF(LOGWARNING, "DXVA will not be used.");
-      return;
-    }
+  CEnumeratorHD enumerator;
+  enumerator.Open(picture.iWidth, picture.iHeight, dxgi_format);
+
+  if (enumerator
+          .SupportedConversions(
+              {picture, IntendToRenderAsHDR(
+                            static_cast<AVColorPrimaries>(picture.color_primaries),
+                            static_cast<AVColorTransferCharacteristic>(picture.color_transfer))})
+          .empty())
```

Comment:
Now it looks worse. If you don't want remove static_casts, leave it as before with the intermediate variable.

My idea was:

```c++
  if (enumerator.SupportedConversions({picture, IntendToRenderAsHDR(picture)}).empty())
  {
    CLog::LogF(LOGWARNING, "DXVA will not be used.");
    return;
  }
```

And in RendererBase:

```c++
bool CRendererBase::IntendToRenderAsHDR(const VideoPicture& picture)
{
  const bool streamIsHDR = (picture.color_primaries == AVCOL_PRI_BT2020) &&
                           (picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
                            picture.color_transfer == AVCOL_TRC_ARIB_STD_B67);

  const bool canDisplayHDR =
      DX::Windowing()->IsHDROutput() || DX::Windowing()->IsHDRDisplaySettingEnabled();

  return streamIsHDR && canDisplayHDR;
}
```

static_cast are not necessary and has not cost pass `VideoPicture&` because is a reference (no object copy)

Reply:
Another way:

```c++
  const SupportedConversionsArgs args = {picture, IntendToRenderAsHDR(picture)};

  if (enumerator.SupportedConversions(args).empty())
  {
    CLog::LogF(LOGWARNING, "DXVA will not be used.");
    return;
  }
```

Reply:
ok I yield ;-) pushed with VideoPicture parameter.
If that's ok I'll squash most commits.